src/ScriptCs.Hosting/ModuleLoader.cs

```diff
@@ -69,7 +71,8 @@ public ModuleLoader(IAssemblyResolver resolver, ILog logger, Action<Assembly, Ag
         {
             if (modulePackagesPaths == null) return;
 
-            if (moduleNames.Length == 1 && DefaultCSharpModules.ContainsKey(moduleNames[0])) // only CSharp module needed
+            // only CSharp module needed - use fast path
+            if (moduleNames.Length == 1 && DefaultCSharpModules.ContainsKey(moduleNames[0]) && (extension.Equals(DefaultCSharpExtension, StringComparison.InvariantCultureIgnoreCase) || string.IsNullOrWhiteSpace(extension))) 
```

Comment:
Or you could swap the order of the conditions around the `||`, but may as well just call the static on `string`.